service/storage_proxy.cc

```diff
@@ -2812,6 +2812,32 @@ storage_proxy::mutate_hint(const schema_ptr& s, const frozen_mutation& m, tracin
     });
 }
 
+std::optional<replica::stale_topology_exception>
+storage_proxy::apply_fence(fencing_token token) const noexcept {
+    const auto fence_version = _shared_token_metadata.get_fence_version();
+    if (!token || token.topology_version >= fence_version) {
+        return std::nullopt;
+    }
+    slogger.warn("Stale topology detected, request has been fenced out, "
```

Comment:
This will cause a huge spam of WARN-level logs if there's a high throughput in writes and we hit this.

Reply:
It's important event, deserves a `WARN` IMHO. I can wrap it in a some rate limiter, I remember it was done somewhere, but can't find an example...

Reply:
Rate limiter sounds good
example usage: https://github.com/scylladb/scylladb/blob/af65d5a1e8e61565da6ef60ae7a4a8dab675e0f6/service/raft/raft_group_registry.cc#L473

Reply:
thanks!